src/main/java/cloud/fogbow/ras/core/ApplicationFacade.java

```diff
@@ -265,6 +267,7 @@ public void deletePublicIp(String publicIpOrderId, String userToken) throws Fogb
                 cloudName = this.cloudListController.getDefaultCloudName();
             }
             RasOperation rasOperation = new RasOperation(Operation.GET_ALL, ResourceType.IMAGE, cloudName);
+            rasOperation.setTargetProvider(providerId);
```

Comment:
Shouldn't the could that sets the provider in case it is null (lines 272-274) be moved before line 270? If so, fix it and check whether this situation appears in other methods of this class.

Reply:
Fixed.